addons/hw_drivers/drivers/SerialScaleDriver.py

```diff
@@ -311,3 +336,48 @@ def supported(cls, device):
         except Exception:
             _logger.exception('Error while probing %s with protocol %s' % (device, protocol.name))
         return False
+
+
+class FAIRBANKSDriver(ScaleDriver):
+    """Driver for the Fairbanks serial scale."""
+    _protocol = FAIRBANKSProtocol
+
+    def __init__(self, device):
+        super().__init__(device)
+        self._device_manufacturer = 'Fairbanks'
+
+    @classmethod
+    def supported(cls, device):
+        """Checks whether the device, which port info is passed as argument, is supported by the driver.
+
+        :param device: path to the device
+        :type device: str
+        :return: whether the device is supported by the driver
+        :rtype: bool
+        """
+
+        protocol = cls._protocol
+        try:
+            with serial_connection(device['identifier'], protocol, is_probing=True) as connection:
+                answer = connection.read(17)
+                STXA = re.search(b'..(\d)*', answer)
+                GW = re.search(protocol.measureRegexp, answer)
+                if STXA and STXA.group() == b'\x021' and GW.group():
+                    return True
+        except serial.serialutil.SerialTimeoutException:
+            pass
+        except Exception:
+            _logger.exception('Error while probing %s with protocol %s' % (device, protocol.name))
+        return False
+
+    def _read_weight(self):
+        """Asks for a new weight from the scale, checks if it is valid and, if it is, makes it the current value."""
+
+        protocol = self._protocol
+        answer = self._connection.read(17)
```

Comment:
It looks very similar to the original read_weight. Shouldn't you simply override the '_get_raw_response'?

Reply:
Maybe but in _read_weight  he make a `connection_write` (so here with a none command) and put in self.data['value'] a float value.